### PR TITLE
Login form validaition

### LINKED ISSUE #20: Do client side validation of the registration constraints
Currently the username field needs to be:
* Longer than 2 characters
* Not longer than 50 chars
* Contain only alpha numerical chars

Moreover the email address needs to be validated client side using the regexp used in the API: https://github.com/ooni/api/blob/master/newapi/ooniapi/auth.py#L254.

We should do client side validation of these parameters and display meaningful error messages to users if the field does not meet these parameters.

### PR DESCRIPTION
**Pull request by sarathms**
Fixes #20 

- Use stricter validation of login form
- Upgrade ooni-components

image

### CODE REVIEW

**Comment by sarathms** [comment]
Use this for preview: [test-lists.test.ooni.org](http://test-lists.test.ooni.org/)

**Comment by hellais** [comment]
> Use this for preview: [test-lists.test.ooni.org](http://test-lists.test.ooni.org/)

When using this preview link, I don't see the validation being done:
Screenshot 2022-04-29 at 13 58 47


When using the deployment preview provided by vercel I get "Network error":
Screenshot 2022-04-29 at 13 59 19

**Review by hellais — COMMENTED** [review]
The code looks good, I am however unable to test it because the deploy link does not work.

File: components/LoginForm.js [hunk 1]
```diff
@@ -24,38 +46,75 @@ export const LoginForm = ({ onLogin }) => {
...
+            minLength: {
+              value: 3,
+              message: 'Should be 3 to 50 characters long'
+            },
+            maxLength: {
+              value: 50,
+              message: 'Should be 3 to 50 characters long'
+            },
+            pattern: {
+              value: /[A-Za-z0-9]+/,
```
Comment (hellais) [inline components/LoginForm.js:104]:
```suggestion
              value: /^[A-Za-z0-9]+$/,
```

**Review by hellais — APPROVED** [review]
Yes the validation works well now. LGTM

**Pull request merged**